src/runtime.rs

```diff
@@ -338,7 +338,11 @@ impl AvailableDataContext {
         } = self;
         let stream_id = *stream_id;
         let (beg, end) = inner.map_read(stream_id)?;
-        let nbytes = unsafe { byte_offset_from(beg, end) };
+        let nbytes = if beg.is_null() || end.is_null() {
+            0
+        } else {
+            unsafe { byte_offset_from(beg, end) }
+        };
```

Comment:
Fwiw, there's a rust feature that got stabilized in rust 1.75 that allows us to potentially replace `bytes_offet_from` with a method in the standard lib (see [docs](https://doc.rust-lang.org/std/primitive.pointer.html#method.byte_offset-1)). Similarly for `byte_offset`.

Those methods don't look like they do any special null handling _([playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=2474100959fcd62b6c93837386890701))_ so you'd still need this null check.

Reply:
10-4. I'll add it in.

Reply:
@nclack Actually it looks like the [Mac](https://github.com/actions/runner-images/blob/macOS-12/20231216.1/images/macos/macos-12-Readme.md#rust-tools) runners are still on 1.74. [Windows](https://github.com/actions/runner-images/blob/win22/20240108.1/images/windows/Windows2022-Readme.md#rust-tools) and [Ubuntu](https://github.com/actions/runner-images/blob/ubuntu22/20240107.1/images/ubuntu/Ubuntu2204-Readme.md#rust-tools) updated only yesterday if you look at the blames. I'll add an issue instead.